cpp/src/reductions/reductions.cpp

```diff
@@ -41,21 +47,27 @@ struct reduce_dispatch_functor {
     case aggregation::SUM:
       return reduction::sum(col, output_dtype, mr, stream);
       break;
+    case aggregation::PRODUCT:
+      return reduction::product(col, output_dtype, mr, stream);
+      break;
     case aggregation::MIN:
       return reduction::min(col, output_dtype, mr, stream);
       break;
     case aggregation::MAX:
       return reduction::max(col, output_dtype, mr, stream);
       break;
+    case aggregation::COUNT_VALID:
```

Comment:
Why is this a reduction?  Analogous to what @jrhemstad said about `nth_element` (it's just `element()`). This is just a property of a column. Has this feature been requested?

Reply:
Not requested. Count valid is still a reduction. Similar reduction call is possible in groupby. so I added here for completeness.